File: O(n²)/bubble_sort.py
def bubble_sort(lista):
    n = len(lista)
    # Laço Externo: Garante que passaremos pela lista n vezes
    for i in range(n):
        # Laço Interno: Compara os pares adjacentes
        # O "- i - 1" é uma pequena otimização, pois os últimos i elementos já estão ordenados
        for j in range(0, n - i - 1):
            # Se o elemento atual for maior que o próximo, troca
            if lista[j] > lista[j + 1]:
                lista[j], lista[j + 1] = lista[j + 1], lista[j]

# Teste
numeros = [64, 34, 25, 12, 22, 11, 90]
bubble_sort(numeros)
print(f"Array ordenado: {numeros}")

File: O(n²)/BubbleSort.java
public class BubbleSort {
    static void bubbleSort(int[] arr) {
        int n = arr.length;
        int temp;
        
        // Laço Externo (i)
        for (int i = 0; i < n - 1; i++) {
            
            // Laço Interno (j) - Processando pares
            for (int j = 0; j < n - i - 1; j++) {
                
                // Comparação e troca
                if (arr[j] > arr[j + 1]) {
                    temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        bubbleSort(arr);
        
        System.out.println("Array ordenado:");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}